app/models/distributed_valid_products.rb

```diff
@@ -0,0 +1,41 @@
+# Finds valid products distributed by a particular distributor in an order cycle
+#
+# If a product without variants is added to an order cycle, and then some
```

Comment:
What do you mean by "product without variants"?

We have product.ensure_standard_variant that keeps products with variants at all the times.

This query returns no results in all AU, UK and FR live.
```
select * from spree_products
where not exists (select 1
                              from spree_variants
                              where product_id = spree_products.id
                                       and is_master = 'f');
```

Reply:
I just moved the comment and honestly, even some the test cases themselves were a bit dubious. I'll review it and make any necessary changes. Thanks for the heads-up!

Reply:
I think this deserves a tech debt issue, to clarify the comment but also the fact that master variants are added to OCs, and why would that ever happen. see my comment below.